extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -125,60 +122,78 @@ export class AmpAdXOriginIframeHandler {
       this.sendEmbedInfo_(this.baseInstance_.isInViewport());
     }));
 
+    // Iframe.onload normally called by the Ad after full load.
+    const iframeLoadPromise = loadPromise(this.iframe).then(() => {
```

Comment:
I went through about a dozen of Ad networks and x20 ad impressions. I had onload arrive earlier than render-start in less than 1% of cases and never more than 10ms earlier. That being said, this code fully supports onload to arrive earlier - just doesn't optimize around it.